src/System.Net.HttpListener/tests/HttpListenerPrefixCollectionTests.cs

```diff
@@ -237,6 +256,227 @@ public void Add_PrefixAlreadyRegisteredAndStarted_ThrowsHttpListenerException()
             }
         }
 
+        public static IEnumerable<object[]> Hosts_TestData()
+        {
+            yield return new object[] { "localhost" };
+            yield return new object[] { "127.0.0.1" };
+
+            if (HttpListenerFactory.SupportsWildcards)
+            {
+                yield return new object[] { "*" };
+                yield return new object[] { "+" };
+            }
+        }
+        
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [MemberData(nameof(Hosts_TestData))]
+        public void Add_SamePortDifferentPathDifferentListenerNotStarted_Works(string host)
+        {
+            var listener1 = new HttpListener();
+            var listener2 = new HttpListener();
+
+            int? freePort = null;
+
+            // Try to find a port that is not being used.
+            for (int port = 1025; port <= IPEndPoint.MaxPort; port++)
+            {
+                string uriPrefix = $"http://{host}:{port}/";
+                try
+                {
+                    listener1.Prefixes.Add(uriPrefix);
+                    Assert.Equal(1, listener1.Prefixes.Count);
+                    listener1.Start();
+
+                    freePort = port;
+                    break;
+                }
+                catch (HttpListenerException)
+                {
+                    // This port is already in use. Skip it and find a port that's not being used.
+                    listener1.Close();
+                    listener1 = new HttpListener();
+                }
+            }
```

Comment:
Could this be factored out into a helper, like:
```C#
public static bool TryGetListenerOnFreePort(out HttpListener listener, out int port);
```
and then you can use that helper here and in the other tests below?